src/xmipp/libraries/reconstruction/ctf_create_ctfdat.cpp

```diff
@@ -130,14 +130,16 @@ class ProgCtfCreateCtfdat : public XmippProgram
 
         size_t id;
 
-        FOR_ALL_OBJECTS_IN_METADATA2(mdIn, mdCtf)
+        auto itIdIn = mdIn.ids().begin();
+        auto itIdCtf = mdIn.ids().begin();
+        for (; itIdIn != mdIn.ids().end(); ++itIdIn, ++itIdCtf)
         {
-            mdIn.getValue(MDL_SELFILE,fnsel,__iter.objId);
-            mdCtf.getValue(MDL_CTF_MODEL,fnctf,__iter2.objId);
+            mdIn.getValue(MDL_SELFILE,fnsel, *itIdIn);
+            mdCtf.getValue(MDL_CTF_MODEL,fnctf, *itIdCtf);
             SFind.read(fnsel);
-            FOR_ALL_OBJECTS_IN_METADATA(SFind)
+            for (size_t objId : SFind.ids())
             {
-                SFind.getValue(MDL_IMAGE,fnimg,__iter.objId);
+                SFind.getValue(MDL_IMAGE,fnimg, *itIdIn);
```

Comment:
please check if it should be `*itIdIn` or `objId`

Reply:
It should be `objId`, thanks for noticing.